src/main/drivers/accgyro_mpu.c

```diff
@@ -286,15 +306,16 @@ mpuDetectionResult_t *mpuDetect(gyroDev_t *gyro)
     bool ack = false;
 #else
     uint8_t sig;
-    bool ack = mpuReadRegisterI2C(MPU_RA_WHO_AM_I, 1, &sig);
+    bool ack = mpuReadRegisterI2C(&gyro->bus, MPU_RA_WHO_AM_I, 1, &sig);
```

Comment:
Agreed. However this PR is not intended to fix all the problems in the old design. As I said, my style is to make incremental changes.